terraform-modules/aws/kms/cloudtrail/main.tf

```diff
@@ -0,0 +1,81 @@
+# This is a standard kms that frees any cloudtrail/trails from vulnerabilities.
+# Docs: https://dev.to/aws-builders/encrypt-cloudtrail-logs-with-multi-region-key-with-terraform-1hln
+
+locals {
+  arn_format  = "arn:${data.aws_partition.current.partition}"
+}
+
+data "aws_partition" "current" {}
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+data "aws_iam_policy_document" "kms" {
+  statement {
+    sid    = "Enable Root User Permissions"
+    effect = "Allow"
+
+    actions = [
+      "kms:Create*",
+      "kms:Describe*",
+      "kms:Enable*",
+      "kms:List*",
+      "kms:Put*",
+      "kms:Update*",
+      "kms:Revoke*",
+      "kms:Disable*",
+      "kms:Get*",
+      "kms:Delete*",
+      "kms:Tag*",
+      "kms:Untag*",
+      "kms:ScheduleKeyDeletion",
+      "kms:CancelKeyDeletion"
+    ]
+
+    #bridgecrew:skip=CKV_AWS_109:This policy applies only to the key it is attached to
+    #bridgecrew:skip=CKV_AWS_111:This policy applies only to the key it is attached to
+    resources = [
+      "*"
+    ]
+
+    principals {
+      type = "AWS"
+
+      identifiers = [
+        "${local.arn_format}:iam::${data.aws_caller_identity.current.account_id}:root"
+      ]
+    }
+  }
+  statement {
+    sid    = "Allow CloudTrail to encrypt  - ${var.cloudtrail_name}"
+    effect    = "Allow"
+    actions   = ["kms:GenerateDataKey*"]
+    resources = ["*"]
+    principals {
+      type        = "Service"
+      identifiers = ["cloudtrail.amazonaws.com"]
+    }
+    condition {
+      test     = "StringLike"
+      variable = "kms:EncryptionContext:aws:cloudtrail:arn"
+      values   = ["arn:aws:cloudtrail:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:trail/${var.cloudtrail_name}"]
+    }
+    condition {
+      test     = "StringEquals"
+      variable = "aws:SourceArn"
+      values   = ["arn:aws:cloudtrail:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:trail/${var.cloudtrail_name}"]
+    }
+  }
+}
+
+resource "aws_kms_alias" "a" {
+  name          = "alias/${var.cloudtrail_name}"
+  target_key_id = aws_kms_key.kms.key_id
+}
+
+resource "aws_kms_key" "kms" {
+  description             = "KMS key for cloudtrail: ${var.cloudtrail_name}"
+  deletion_window_in_days = 10
+  enable_key_rotation     = true
```

Comment:
Since you are parameterizing the `deletion_window_in_days` probably just go ahead and parameterize this as well.

Reply:
I've refactor and parameterized here: https://github.com/ManagedKube/kubernetes-ops/pull/362/files#diff-4021d6e58452d498e5f201480e99bb7aba201fc118975760309c5c6d2b404637R7